PocketKit/Sources/Textile/Style/PocketButtonStyle.swift

```diff
@@ -57,73 +66,90 @@ public struct PocketButtonStyle: ButtonStyle {
         }
     }
 
-    public init(_ variation: Variation) {
+    public init(_ variation: Variation, _ size: Size = .normal) {
         self.variation = variation
+        self.size = size
     }
 
     public func makeBody(configuration: Configuration) -> some View {
-        PocketButton(configuration: configuration, variation: self.variation)
+        PocketButton(configuration: configuration, variation: self.variation, size: self.size)
     }
 
     struct PocketButton: View {
         let configuration: ButtonStyle.Configuration
 
         let variation: Variation
+        let size: Size
 
         @Environment(\.isEnabled)
         private var isEnabled: Bool
 
         var body: some View {
```

Comment:
This function is massive and very hard to read.
I'll post some suggestions inline.

Reply:
thanks! I agree, I added a TODO to look into refactoring our Buttons to match our Figma files at a later date